### PR TITLE
Support non string values

### PR DESCRIPTION
**Pull request by caiogondim**
Should fix https://github.com/caiogondim/logdown.js/issues/50

### CODE REVIEW

**Comment by caiogondim** [comment]
It's seems to be broken.
I fixed the tests and broken the feature =/

Gimme a couple minutes.

**Comment by caiogondim** [comment]
Good to go

**Comment by bennyn** [comment]
I tested your adjustments today. In the Browser it looks good for me but in a Node.js application it still stringifies objects.

## Results

### Browser

capture

### Node.js

capture-2

For Node.js applications I would expect something like this (which is what `console.log` outputs on Node):

capture

My test code:

```javascript
var Logdown = require('logdown');

var options = {
  alignOutput: true,
  prefix: 'AwesomeLoggerRocker'
};

var AwesomeLoggerRocker = new Logdown(options);

function log(instance) {
  var methods = [
    'debug',
    'error',
    'info',
    'log',
    'warn'
  ];


  var number = 72;

  var object = {
    name: 'Arnold',
    age: 69
  };

  var text = 'Hello World';

  methods.forEach(function(method) {
    instance[method](text, number);
  });

  methods.forEach(function(method) {
    instance[method](text, object);
  });


  methods.forEach(function(method) {
    instance[method](text);
  });
};

log(AwesomeLoggerRocker);
```

**Comment by caiogondim** [comment]
I indeed only changed for the browser.
Will make a patch for NodeJS support.

**Comment by bennyn** [comment]
Ok. So we will merge this one here as "minor" (1.5) and then do a seperate PR for Node.js which will be "1.5.1"?

**Comment by caiogondim** [comment]
No. I will make a patch to support Node as well.
Im about to get into the plane.
Should send the code by tomorrow =)

**Comment by bennyn** [comment]
Okay! Have a good flight and take care! 🛩 😄

**Comment by caiogondim** [comment]
Sorry for the delay on this issue.
Gimme a couple more days 😊.
I'm on it.

**Comment by bennyn** [comment]
@caiogondim Ok! I am also back from vacation in Brussels. Ready to review it anytime. Ok, let's say "almost" anytime. 😁

**Comment by bennyn** [comment]
Hey @caiogondim, can we please get this fix in (even if it just works for browsers)? I am eager to use this one already in the browser and it's idling since 20 days already...

A timely release would be nice because I am using Logdown in a web application where I need it for debugging in the Browser but I get log messages like this:

> cryptobox.store.IndexedDB Loaded record "65535" from object store "prekeys". [object Object] 

Merging your PR would improve the current situation because it would turn the String `[object Object]` into a real object (for Browsers).

You could make a second PR for having the same feature on Node.js.

**Comment by caiogondim** [comment]
Sorry for the delay.
I'm trying to avoid multi-tasking on projects.
Was about to finish one to then jump into `logdown.js`.

Due to the urgency I will jump into this to support Node.

Will keep you up to date.

**Comment by caiogondim** [comment]
Working.

Can you check?

**Comment by bennyn** [comment]
I tried my test code but it fails. 

It reports: 

> TypeError: CreateListFromArrayLike called on non-object at Logdown.(anonymous function) [as debug] (D:\dev\projects\caiogondim\logdown.js\dist\index.js:168:43)

The line it complains about is:

```javascript
        ;(console[method] || console.log).apply(
          console,
          preparedOutput
        )
```

Looks like it doesn't like the `apply` on it.

Here is my test code:

**dist/test.js**

```javascript
var Logdown = require('./index');

var AwesomeLoggerRocker = new Logdown({
    alignOutput: true,
    prefix: 'AwesomeLoggerRocker'
});

var text = 'Hello World';
var object = {
    name: 'Arnold',
    age: 69
};

AwesomeLoggerRocker.log(text, object);
```

**Comment by caiogondim** [comment]
Will take a look.

**Comment by bennyn** [comment]
It does not complain, when I surround `preparedOutput` in an array. So with this snippet my test code runs through:

```javascript
        ;(console[method] || console.log).apply(
          console,
          [preparedOutput]
        )
```

But then, the output of `AwesomeLoggerRocker.log(text, object);` is just `3` (which is wrong).

I found a nice blog post on the `CreateListFromArrayLike` error here: http://jagkanjavascript.blogspot.de/2016/04/indirectly-calling-functions-using-call.html

**Comment by caiogondim** [comment]
It was my bad.
I can't never remember when `Array` methods returns the modified array or whatever.

`Array.prototype.push` returns the new length ¯\_(ツ)_/¯

**Comment by caiogondim** [comment]
Will fix tests as well.
Was using ES2015+ syntax

**Comment by caiogondim** [comment]
Good to go.
Can you check now?

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,9 @@ os:
   - linux
 language: node_js
 node_js:
-  - "6.9"
-  - "4.6"
+  - 4
+  - 5
```
Comment (bennyn) [inline .travis.yml]: If we want to only support [Node.js LTS versions](https://github.com/nodejs/LTS#lts-schedule), then we should drop testing against Node v5.

File: test/browser/logging-methods.js [hunk 2]
```diff
@@ -7,7 +7,8 @@
   sinon.assert.expose(chai.assert, {prefix: ''})
   var assert = chai.assert
 
-  var methods = ['debug', 'log', 'info', 'warn', 'error']
+  // var methods = ['debug', 'log', 'info', 'warn', 'error']
```
Comment (bennyn) [inline test/browser/logging-methods.js]: Comment can be dropped!?

File: src/index.js [hunk 3]
```diff
@@ -233,7 +190,7 @@
 
       if (isBrowser()) {
         styles.push(match.rule.style)
-        styles.push('color:inherit;')
+        styles.push('') // Empty string resets style.
```
Comment (bennyn) [inline src/index.js]: Should we document `// Empty string resets style.` somewhere else? Because setting `''` happens now in multiple parts of the code.

**Comment by bennyn** [comment]
@caiogondim **PERFECT!!!** For me it works now in the Browser and on Node.js. 🎉 I left some comments in my code review, so you can go ahead. 🤗 🤗

File: test/browser/logging-methods.js (hunk 2 above)
Comment (caiogondim) [inline test/browser/logging-methods.js]: Nice catch.

File: src/index.js (hunk 3 above)
Comment (caiogondim) [inline src/index.js]: But `styles.push('') `happens only once. The others are unrelated.
But if it's not clear, let me know.
If the code itself is not clear, we have to put comments.

**Comment by caiogondim** [comment]
Care to merge and make a new release? 🎉

File: src/index.js (hunk 3 above)
Comment (bennyn) [inline src/index.js]: Ah, I see! Than take this one as **solved**. 💃

File: .travis.yml (hunk 1 above)
Comment (caiogondim) [inline .travis.yml]: Fixed in 44ac46ed0552e0a8550826c6e7357df6e05e9288

**Comment by bennyn** [comment]
Happy to do it! 1.5.0?

**Comment by caiogondim** [comment]
👍

**Comment by bennyn** [comment]
Done! https://github.com/caiogondim/logdown.js/releases/tag/v1.5.0 🎉 

So happy that it's there now. Thank you!! 👍

**Pull request merged**